x509-identity-mgmt/ejbca/bin/dojot-custom/configuration-variables.sh

```diff
@@ -0,0 +1,186 @@
+#                      before the old CRL expires. The default value
+#                      is 10 minutes, meaning that if the CRL Expire
+#                      period is 24 hours, a new CRL will be issued
+#                      after 23h50m.
+# Delta CRL Period === The validity period for generated delta CRLs
+#                      if delta CRLs are issued. Delta CRLs are only
+#                      issued if this period is larger than 0.
+#
+# "X509 Identity CA"
+readonly CRL_EXPIRE_PERIOD_DEVICES_CA=$(convToMillis "${CRL_EXPIRE_PERIOD_DEVICES_CA:-1d}")
+readonly CRL_ISSUE_INTERVAL_DEVICES_CA=$(convToMillis "${CRL_ISSUE_INTERVAL_DEVICES_CA:-0m}")
+readonly CRL_OVERLAP_TIME_DEVICES_CA=$(convToMillis "${CRL_OVERLAP_TIME_DEVICES_CA:-10m}")
+readonly DELTA_CRL_PERIOD_DEVICES_CA=$(convToMillis "${DELTA_CRL_PERIOD_DEVICES_CA:-1h}")
+#
+# "Services CA"
+readonly CRL_EXPIRE_PERIOD_SERVICES_CA=$(convToMillis "${CRL_EXPIRE_PERIOD_SERVICES_CA:-1d}")
+readonly CRL_ISSUE_INTERVAL_SERVICES_CA=$(convToMillis "${CRL_ISSUE_INTERVAL_SERVICES_CA:-0m}")
+readonly CRL_OVERLAP_TIME_SERVICES_CA=$(convToMillis "${CRL_OVERLAP_TIME_SERVICES_CA:-10m}")
+readonly DELTA_CRL_PERIOD_SERVICES_CA=$(convToMillis "${DELTA_CRL_PERIOD_SERVICES_CA:-1h}")
+
+
+# *-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+# ------------------------- EJBCA Services --------------------------
+# The CRL Updater checks if any of the configured CAs need a new CRL
+# and generates it if necessary. The worker has no additional settings
+# and only supports the periodical interval.
+# For more details:
+# https://doc.primekey.com/ejbca6152/ejbca-operations/ejbca-concept-guide/services#Services-CRL_Updater_Service
+readonly CRL_UPDATER_SERVICE_NAME=${CRL_UPDATER_SERVICE_NAME:-"CRL Updater"}
+readonly CRL_UPDATER_SERVICE_INTERVAL_VALUE=${CRL_UPDATER_SERVICE_INTERVAL_VALUE:-"30"}
+readonly CRL_UPDATER_SERVICE_INTERVAL_UNIT=${CRL_UPDATER_SERVICE_INTERVAL_UNIT:-"MINUTES"}
+#
+# The renew CA service can be used to automatically renew CAs that are about to expire.
+# For more details:
+# https://doc.primekey.com/ejbca6152/ejbca-operations/ejbca-concept-guide/services#Services-RenewCAService
+readonly CA_RENEWER_SERVICE_NAME=${CA_RENEWER_SERVICE_NAME:-"CAs Renewer"}
+readonly CA_RENEWER_SERVICE_INTERVAL_VALUE=${CA_RENEWER_SERVICE_INTERVAL_VALUE:-"1"}
+readonly CA_RENEWER_SERVICE_INTERVAL_UNIT=${CA_RENEWER_SERVICE_INTERVAL_UNIT:-"DAYS"}
+readonly CA_RENEWER_SERVICE_TIME_BEFORE_EXP=${CA_RENEWER_SERVICE_TIME_BEFORE_EXP:-"10"}
+readonly CA_RENEWER_SERVICE_TIME_BEFORE_EXP_UNIT=${CA_RENEWER_SERVICE_TIME_BEFORE_EXP_UNIT:-"DAYS"}
+
+
+# *-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+# ------------------------- TLS APP SERVER --------------------------
+# The host and domain are defined when the container is initialized.
+# In this case the parameters "hostname" and "domainname" must be
+# defined in docker-compose.yml
```

Comment:
must be defined in environment variables?

Reply:
In this case, the value for the `hostname` and `domainname` must be declared (with the same name) from the docker-compose.yml file.

See for example lines 9 and 10 of this file: [docker-compose.yml](https://github.com/dojot/dojot/pull/1469/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.sh&file-filters%5B%5D=.xml&file-filters%5B%5D=.yml#diff-f9a0ddfcc35b4e70ef67980c8f4eca18)

Reply:
I think **maybe** It's better pass everything by environment variables, I don't know.
I don't know how it works in k8s, I think maybe @andersonluisribeiro can give an opinion on that.
In addition, it is better not to mention the docker-compose, since deployment can happen in k8s for example.

Reply:
Hi,I think that environment variables it's better for k8s.

Reply:
I tried to implement it according to your suggestion, but EJBCA makes use of the "hostname" command of linux to generate the certificate of the application server that it runs on. When the container does not have a defined hostname and domainname, a random value is assumed and the EJBCA generates a self-signed certificate for the Wildfly server. This impacts the communication of the NodeJS application that runs in the same container and has to communicate with the EJBCA via the web service in TLS.

@mprevide suggested checking the following links to reproduce this docker-compose configuration on kubernetes:

https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods-hostname-and-subdomain-fields